src/Microsoft.Azure.SignalR.Management/IServiceManager.cs

```diff
@@ -32,15 +42,30 @@ public interface IServiceManager : IDisposable
         /// <param name="claims">The claim list to be put into access token.</param>
         /// <param name="lifeTime">The lifetime of the token. The default value is one hour.</param>
         /// <returns>Client access token to Azure SignalR Service.</returns>
+        /// <remarks>Please use <see cref="GetEndpointAccessInfoAsync(string, HttpContext, string, IList{Claim}, TimeSpan?)"/> instead.</remarks>
+        [Obsolete("Use method GetEndpointAccessInfo instead.")]
         string GenerateClientAccessToken(string hubName, string userId = null, IList<Claim> claims = null, TimeSpan? lifeTime = null);
 
         /// <summary>
         /// Creates an client endpoint for SignalR hub connections to connect to Azure SignalR Service
         /// </summary>
         /// <param name="hubName">The hub name.</param>
         /// <returns>Client endpoint to Azure SignalR Service.</returns>
+        /// <remarks>Please use <see cref="GetEndpointAccessInfoAsync(string, HttpContext, string, IList{Claim}, TimeSpan?)"/> instead.</remarks>
+        [Obsolete("Use method GetEndpointAccessInfo instead.")]
```

Comment:
can we only have interface changes when everything else is ready?

Reply:
Negotiation part removed.